src/components/Datatable/Datatable.tsx

```diff
@@ -134,17 +140,28 @@ const Datatable = <D extends Record<string, unknown>>({
     [onDataFetch, appliedFilters, searchQuery],
   );
 
+  const handleOnSuggestionsFetch = useCallback(
+    async (query: string) => {
+      setFilterSuggestions(await onSuggestionsFetch(query));
+    },
+    [onSuggestionsFetch],
+  );
+
   const handleOnSearch = useCallback(
-    (query: string) => {
-      setSearchQuery(query);
-      onDataFetch({
-        pageSize: defaultPageSize,
-        pageIndex: 0,
-        query,
-        filters: appliedFilters,
-      });
+    async (queryValue: string) => {
```

Comment:
why `async` there is no `await` in the function

Reply:
why did you change it back to `queryValue`?

Reply:
the await is on L147

Reply:
but that's is different function, no need to have async here